etna/api/urls.py

```diff
@@ -1,16 +1,134 @@
+        }
+        for restriction in restrictions:
+            if restriction.restriction_type == PageViewRestriction.PASSWORD:
+                if "password" in request.GET:
+                    if constant_time_compare(
+                        request.GET["password"], restriction.password
+                    ):
+                        return Response(data)
+                    else:
+                        data = restricted_data | {"message": "Incorrect password."}
+                        return Response(data)
+                else:
+                    data = restricted_data | {
+                        "message": "Password required to view this resource.",
+                    }
+                    return Response(data)
+        data = restricted_data | {
+            "message": "Selected privacy mode is not compatible with this API.",
+        }
+        return Response(data, status=status.HTTP_403_FORBIDDEN)
+
+    def get_base_queryset(self):
+        """
+        Returns a queryset containing all pages that can be seen by this user.
+
+        This is used as the base for get_queryset and is also used to find the
+        parent pages when using the child_of and descendant_of filters as well.
+        """
+
+        request = self.request
+
+        # Get all live pages
+        queryset = Page.objects.all().live()
+
+        # Check if we have a specific site to look for
+        if "site" in request.GET:
+            # Optionally allow querying by port
+            if ":" in request.GET["site"]:
+                (hostname, port) = request.GET["site"].split(":", 1)
+                query = {
+                    "hostname": hostname,
+                    "port": port,
+                }
+            else:
+                query = {
+                    "hostname": request.GET["site"],
+                }
+            try:
+                site = Site.objects.get(**query)
+            except Site.MultipleObjectsReturned:
+                raise BadRequestError(
+                    "Your query returned multiple sites. Try adding a port number to your site filter."
+                )
+        else:
+            # Otherwise, find the site from the request
+            site = Site.find_for_request(self.request)
+
+        if site:
+            base_queryset = queryset
+            queryset = base_queryset.descendant_of(site.root_page, inclusive=True)
+
+            # If internationalisation is enabled, include pages from other language trees
+            if getattr(settings, "WAGTAIL_I18N_ENABLED", False):
+                for translation in site.root_page.get_translations():
+                    queryset |= base_queryset.descendant_of(translation, inclusive=True)
+
+        else:
+            # No sites configured
+            queryset = queryset.none()
+
+        return queryset
+
+    # TODO: Why do I have to redefine these after meta_fields in etna/core/models/basepage.py?
```

Comment:
???

Reply:
I thought it might've been because you're querying on `Page` rather than `BasePage` in line 89, but that doesn't work because `BasePage` is abstract. I tried it with `ArticlePage` and same thing there - no `meta_fields` ☹️ 

Did `meta_fields` ever work on `BasePage` with the default serializer? I tried adding it to what we have on develop and it made no difference...

Reply:
You're right - they don't look like they work on abstract classes.

Kept the fields in the API but removed from `BasePage`.